sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/SecretClientImpl.java

```diff
@@ -0,0 +1,934 @@
+            .setValue(secret.getValue())
+            .setTags(secret.getProperties().getTags())
+            .setContentType(secret.getProperties().getContentType())
+            .setSecretAttributes(new SecretRequestAttributes(secret.getProperties()));
+        return parameters;
+    }
+
+    /**
+     * Creates a new secret in the key vault asynchronously.
+     * @param name the secret of the secret.
+     * @param value the value of the secret.
+     * @param context the context to use while executing request.
+     * @return A {@link Mono} containing the Response with created secret.
+     */
+    public Mono<Response<KeyVaultSecret>> setSecretWithResponseAsync(String name, String value, Context context) {
+        SecretRequestParameters parameters = new SecretRequestParameters().setValue(value);
+        return service.setSecretAsync(vaultUrl, name, apiVersion, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
+            .doOnRequest(ignored -> logger.verbose("Setting secret - {}", name))
+            .doOnSuccess(response -> logger.verbose("Set secret - {}", response.getValue().getName()))
+            .doOnError(error -> logger.warning("Failed to set secret - {}", name, error));
+    }
+
+    /**
+     * Creates a new secret in the key vault.
+     * @param name the secret of the secret.
+     * @param value the value of the secret.
+     * @param context the context to use while executing request.
+     * @return the Response with created secret.
+     */
+    public Response<KeyVaultSecret> setSecretWithResponse(String name, String value, Context context) {
+        SecretRequestParameters parameters = new SecretRequestParameters().setValue(value);
+        enableSyncRestProxy(context);
+        return service.setSecret(vaultUrl, name, apiVersion, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE));
+    }
+
+    public Mono<Response<KeyVaultSecret>> getSecretWithResponseAsync(String name, String version, Context context) {
+        context = context == null ? Context.NONE : context;
+        return service.getSecretAsync(vaultUrl, name, version == null ? "" : version, apiVersion, ACCEPT_LANGUAGE,
+                CONTENT_TYPE_HEADER_VALUE, context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
+            .doOnRequest(ignoredValue -> logger.verbose("Retrieving secret - {}", name))
+            .doOnSuccess(response -> logger.verbose("Retrieved secret - {}", response.getValue().getName()))
+            .doOnError(error -> logger.warning("Failed to get secret - {}", name, error));
+    }
+
+    public Response<KeyVaultSecret> getSecretWithResponse(String name, String version, Context context) {
+        context = context == null ? Context.NONE : context;
+        enableSyncRestProxy(context);
```

Comment:
does this work ? `Context` is immutable.
Should this be `context =  enableSyncRestProxy(context);` ?
Please add tests that assert that we're hitting right api on the http client layer (@samvaity is doing that in app config please sync with her).